core/templates/pages/learner-dashboard-page/new-learner-dashboard.component.ts

```diff
@@ -0,0 +1,388 @@
+    private urlService: UrlService
+  ) {}
+
+  ngOnInit(): void {
+    this.loaderService.showLoadingScreen('Loading');
+
+    let userInfoPromise = this.userService.getUserInfoAsync();
+    userInfoPromise.then(userInfo => {
+      const username = userInfo.getUsername();
+      if (username) {
+        this.username = username;
+        [this.profilePicturePngDataUrl, this.profilePictureWebpDataUrl] = (
+          this.userService.getProfileImageDataUrl(username));
+      } else {
+        this.profilePictureWebpDataUrl = (
+          this.urlInterpolationService.getStaticImageUrl(
+            AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH));
+        this.profilePicturePngDataUrl = (
+          this.urlInterpolationService.getStaticImageUrl(
+            AppConstants.DEFAULT_PROFILE_IMAGE_PNG_PATH));
+      }
+    });
+    this.homeImageUrl = this.getStaticImageUrl('/learner_dashboard/home.svg');
+    this.todolistImageUrl = this.getStaticImageUrl(
+      '/learner_dashboard/todolist.svg');
+    this.progressImageUrl = this.getStaticImageUrl(
+      '/learner_dashboard/progress.svg');
+
+    let dashboardTopicAndStoriesDataPromise = (
+      this.learnerDashboardBackendApiService
+        .fetchLearnerDashboardTopicsAndStoriesDataAsync());
+    dashboardTopicAndStoriesDataPromise.then(
+      responseData => {
+        this.completedStoriesList = (
+          responseData.completedStoriesList);
+        this.learntTopicsList = (
+          responseData.learntTopicsList);
+        this.partiallyLearntTopicsList = (
+          responseData.partiallyLearntTopicsList);
+        this.topicsToLearn = (
+          responseData.topicsToLearnList);
+        this.untrackedTopics = (
+          responseData.untrackedTopics);
+        this.allTopics = (
+          responseData.allTopicsList);
+        this.learntToPartiallyLearntTopics = (
+          responseData.learntToPartiallyLearntTopics);
+        this.activeSubsection = (
+          LearnerDashboardPageConstants
+            .LEARNER_DASHBOARD_SUBSECTION_I18N_IDS.SKILL_PROFICIENCY
+        );
+      }, errorResponseStatus => {
+        if (
+          AppConstants.FATAL_ERROR_CODES.indexOf(errorResponseStatus) !== -1) {
+          this.alertsService.addWarning(
+            'Failed to get learner dashboard topics and stories data');
+        }
+      }
+    );
+
+    let learnerGroupFeatureIsEnabledPromise = (
+      this.learnerGroupBackendApiService.isLearnerGroupFeatureEnabledAsync()
+    );
+    learnerGroupFeatureIsEnabledPromise.then(featureIsEnabled => {
+      this.LEARNER_GROUP_FEATURE_IS_ENABLED = featureIsEnabled;
+      this.learnerDashboardBackendApiService
+        .fetchLearnerDashboardLearnerGroupsAsync().then(
+          (learnerDashboardLearnerGroups) => {
+            let learnerGroupsJoined = (
+              learnerDashboardLearnerGroups.learnerGroupsJoined);
+            for (var learnerGroup of learnerGroupsJoined) {
```

Comment:
```suggestion
            for (let learnerGroup of learnerGroupsJoined) {
```
use let on the place of var. here and elsewhere